### DISCUSSION TITLE
Session screenshot

### DISCUSSION THREAD

**Discussion by liquidbydesign** [1Remote/1Remote#450] [Ideas]
I often connect to VNC servers to take screenshots.
Right now I have to take a screenshot in Windows then open Paint to edit the image to just contain the active VNC area without the interface and desktop.
It would be great to have a screenshot feature like TeamViewer has: a button or key combination opens a popup where you can choose to either save the active area to a PNG-file or copy to clipboard. The active area should always be saved or copied in native resoultion (i.e. 640x480, 1920x1080 or what your VNC server has) even if you have enabled resolution streching in 1RM.

**Comment by majkinetor** [comment]
How about ALT+PRTSCREEN (screenshot active window) on the host while 1RM is "focused"? Not exactly the same as it will catch 1RM tabs too but almost the same. 

I doubt 1RM can take a full resolution screenshot efficiently when in window, I beleive that would require it to launch a new window in native resolution, somewhere hidden, and then take a screenshot, since you can have smaller window to remote side generally speaking. Its not the same as Team Viewer as it has itself running on the both sides, unlike 1RM.

Other then that, it does seem like a solid and generally useful feature. If it is done, you could even call it for all open tabs in sequence to produce multiple screenshots.

**Comment by liquidbydesign** [comment]
Even if 1RM hides border/tabs the ALT+PRTSCREEN method would catch the black border around the active area if you haven't sized the window exactly right. mRemoteNG has this printscreen feature that I'm looking for, and it works almost perfect. The only thing I'm missing is an additional choice to copy to clipboard instead of saving. Unfortunately mRemoteNG VNC client is very buggy and disconnects every few seconds, that's why I stumbled across 1RM.

<details><summary>Details</summary>
<p>
<img width="1081" alt="VNC Screenshot" src="https://github.com/1Remote/1Remote/assets/72623992/76ca4279-5898-4813-967f-8267b2912808">
</p>
</details>

**Reply by majkinetor** [reply]
This is how ALT + PRTSCREEN looks when I do it.

<details><summary>details</summary> 

![image](https://github.com/1Remote/1Remote/assets/85767/b5674334-4bf1-4db6-a4f5-6559864eaf6e)
</details>

**Reply by liquidbydesign** [reply]
This is how ALT + PRTSCREEN looks for me. I want to get rid of the black area and the interface without the need to manually edit in paint.

<details><summary>Details</summary>
<p>
<img width="767" alt="image" src="https://github.com/1Remote/1Remote/assets/72623992/07d57bfe-2bbf-4e69-870b-a79ceff66a5f">
</p>
</details>

**Reply by VShawn** [reply]
Technically, we should be able to provide screenshots of the built-in protocol. But, I'm not sure if the screen can be captured correctly when using a third-party VNC client(as shown below).

![image](https://github.com/1Remote/1Remote/assets/10143738/8220e314-4023-4708-85ec-da5ac3544c40)

**Reply by liquidbydesign** [reply]
I'm only using the build-in VNC client so that's fine.
